WhisperAPI/WhisperAPI/Services/Suggestions/SuggestionsService.cs

```diff
@@ -144,6 +145,35 @@ public bool UpdateContextWithSelectedSuggestion(ConversationContext conversation
             return false;
         }
 
+        internal async Task<IEnumerable<Recommendation<Document>>> GetQuerySearchRecommendations(ConversationContext conversationContext)
+        {
+            var allParsedRelevantQueries = conversationContext.ContextItems
+                .Where(c => c.SearchQuery.Type == SearchQuery.MessageType.Customer && c.Relevant)
+                .Select(x => x.NlpAnalysis.ParsedQuery).ToList();
+
+            var allWords = string.Join(" ", allParsedRelevantQueries.ToArray()).Split(" ").AsEnumerable();
+            var query = string.Join(" ", allWords.Reverse().Take(7));
```

Comment:
Do the reverse on parsed queries, this avoids to do a ToList() on them. Also, we assume first words of a parsed query are more relevant.